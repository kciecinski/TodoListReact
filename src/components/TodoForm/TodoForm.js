import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import { MDBBtn, MDBIcon, MDBInput } from 'mdbreact';


export default function TodoForm({ onAddItem }) {
  const [todoText, setTodoText] = useState('');
  const inputEl = useRef(null);

  const submitForm = (text) => {
    if (text !== '') {
      onAddItem(text);
      setTodoText('');
      if (inputEl.current) inputEl.current.state.innerValue = '';
    }
  };

  const handleKey = (e) => {
    if (e.which === 13) {
      submitForm(todoText);
    }
  };

  return (
    <div>
      <MDBInput
        onKeyPress={handleKey}
        size="lg"
        ref={inputEl}
        label="Add Tasks"
        onChange={e => setTodoText(e.target.value)}
        className="todo_form"
      />
      <MDBBtn className="btn_circle" color="pink" aria-label="Add" onClick={() => submitForm(todoText)} type="submit">
        <MDBIcon icon="plus" size="2x" />
      </MDBBtn>
    </div>
  );
}
TodoForm.propTypes = {
  onAddItem: PropTypes.func.isRequired,
};
