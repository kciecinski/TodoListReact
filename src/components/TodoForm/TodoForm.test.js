import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TodoForm from './TodoForm';

configure({ adapter: new Adapter() });

describe('<TodoForm />', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      onAddItem: jest.fn(),
    };
    wrapper = shallow(<TodoForm {...props} />);
  });

  it('calls onAddItem correctly on click', () => {
    wrapper.find('Input').simulate('change', { target: { value: 'Buy Stuff' } });
    wrapper.find('Button').simulate('click');

    expect(props.onAddItem).toHaveBeenCalledWith('Buy Stuff');
  });

  it('dont calls onAddItem when empty', () => {
    wrapper.find('Button').simulate('click');

    expect(props.onAddItem).not.toHaveBeenCalledWith('Buy Stuff');
  });

  it('calls onAddItem correctly on pressing enter', () => {
    wrapper.find('Input').simulate('change', { target: { value: 'Buy Other Stuff' } });
    wrapper.find('Input').simulate('keypress', { which: 13 });

    expect(props.onAddItem).toHaveBeenCalledWith('Buy Other Stuff');
  });

  it('renders correctly with defoult props', () => {
    const defoultWrapper = shallow(<TodoForm />);
    expect(toJson(defoultWrapper)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
