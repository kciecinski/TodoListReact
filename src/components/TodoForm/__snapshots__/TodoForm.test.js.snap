// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> renders correctly 1`] = `
<div>
  <Input
    className="todo_form"
    containerClass=""
    disabled={false}
    error=""
    filled={false}
    gap={false}
    group={false}
    icon=""
    iconBrand={false}
    iconClass=""
    iconLight={false}
    iconRegular={false}
    label="Add Tasks"
    labelClass=""
    onChange={[Function]}
    onKeyPress={[Function]}
    outline={false}
    size="lg"
    success=""
    tag="input"
    type="text"
    validate={false}
    valueDefault=""
  />
  <Button
    aria-label="Add"
    className="btn_circle"
    color="pink"
    onClick={[Function]}
    tag="button"
    type="submit"
  >
    <Fa
      border={false}
      brand={false}
      className=""
      fab={false}
      fal={false}
      far={false}
      fixed={false}
      flip=""
      icon="plus"
      inverse=""
      light={false}
      list={false}
      pull=""
      pulse={false}
      regular={false}
      rotate=""
      size="2x"
      spin={false}
      stack=""
    />
  </Button>
</div>
`;

exports[`<TodoForm /> renders correctly with defoult props 1`] = `
<div>
  <Input
    className="todo_form"
    containerClass=""
    disabled={false}
    error=""
    filled={false}
    gap={false}
    group={false}
    icon=""
    iconBrand={false}
    iconClass=""
    iconLight={false}
    iconRegular={false}
    label="Add Tasks"
    labelClass=""
    onChange={[Function]}
    onKeyPress={[Function]}
    outline={false}
    size="lg"
    success=""
    tag="input"
    type="text"
    validate={false}
    valueDefault=""
  />
  <Button
    aria-label="Add"
    className="btn_circle"
    color="pink"
    onClick={[Function]}
    tag="button"
    type="submit"
  >
    <Fa
      border={false}
      brand={false}
      className=""
      fab={false}
      fal={false}
      far={false}
      fixed={false}
      flip=""
      icon="plus"
      inverse=""
      light={false}
      list={false}
      pull=""
      pulse={false}
      regular={false}
      rotate=""
      size="2x"
      spin={false}
      stack=""
    />
  </Button>
</div>
`;
